### PR TITLE
[Bug] LocalStorageChecker should support MEMORY_LOCALFILE and MEMORY_LOCALFILE_HDFS

### PR DESCRIPTION
**Pull request by jerqi**
### What changes were proposed in this pull request?
Add a method containsLocalFile that judge storageType use localfile, and LOCALFILE, LOCALFILE_HDFS,MEMORY_LOCALFILE,MEMORY_LOCALFILE_HDFS will return true.

### Why are the changes needed?
If we don't have this patch, MEMORY_LOCALFILE and MEMORY_LOCALFILE_HDFS can't use LocalStorageChecker.

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
Manually.

### CODE REVIEW

File: storage/src/main/java/com/tencent/rss/storage/util/ShuffleStorageUtils.java [hunk 1]
```diff
@@ -239,4 +239,15 @@ public static long uploadFile(File file, HdfsFileWriter writer, int bufferSize)
       throw new IOException(e);
     }
   }
+
+  public static boolean containsLocalFile(String storageType) {
```
Comment (colinmjj) [inline storage/src/main/java/com/tencent/rss/storage/util/ShuffleStorageUtils.java:243]: return StorageType.LOCALFILE.name().equals(storageType)
        || StorageType.LOCALFILE_HDFS.name().equals(storageType)
        || StorageType.LOCALFILE_HDFS_2.name().equals(storageType)
        || StorageType.MEMORY_LOCALFILE.name().equals(storageType)
        || StorageType.MEMORY_LOCALFILE_HDFS.name().equals(storageType);

Comment (jerqi) [reply storage/src/main/java/com/tencent/rss/storage/util/ShuffleStorageUtils.java:243]: OK.

File: common/src/main/java/com/tencent/rss/common/util/RssUtils.java [hunk 2]
```diff
@@ -161,29 +162,33 @@ public static Roaring64NavigableMap deserializeBitMap(byte[] bytes) throws IOExc
...
+        if (bufferOffset >= readBufferSize) {
+          ShuffleDataSegment sds = new ShuffleDataSegment(fileOffset, bufferOffset, bufferSegments);
+          dataFileSegments.add(sds);
+          bufferSegments = Lists.newArrayList();
+          bufferOffset = 0;
+          fileOffset = -1;
+        }
+      } catch (BufferUnderflowException ue) {
+        LOGGER.warn("Read index data under flow", ue);
+        break;
```
Comment (colinmjj) [inline common/src/main/java/com/tencent/rss/common/util/RssUtils.java:191]: here is for bug fix with broken index data?

Comment (jerqi) [reply common/src/main/java/com/tencent/rss/common/util/RssUtils.java:191]: Yes

Comment (jerqi) [reply common/src/main/java/com/tencent/rss/common/util/RssUtils.java:191]: I separate them. It should be another pr.

**Review by colinmjj — APPROVED** [review]
LGTM

**Pull request merged**